packages/atomic/src/components/commerce/facets/atomic-commerce-timeframe-facet/atomic-commerce-timeframe-facet.tsx

```diff
@@ -0,0 +1,317 @@
+    return !this.shouldRenderFacet;
+  }
+
+  private resetRange() {
+    this.inputRange = undefined;
+    this.facet.setRanges([]);
+  }
+
+  private registerFacetToStore() {
+    const facetInfo: FacetInfo = {
+      label: () => this.bindings.i18n.t(this.displayName),
+      facetId: this.facet.state?.facetId,
+      element: this.host,
+      isHidden: () => this.isHidden,
+    };
+
+    this.bindings.store.registerFacet('dateFacets', {
+      ...facetInfo,
+      format: (value) => this.formatFacetValue(value),
+    });
+
+    initializePopover(this.host, {
+      ...facetInfo,
+      hasValues: () => this.hasValues,
+      numberOfActiveValues: () => this.numberOfSelectedValues,
+    });
+  }
+
+  private formatFacetValue(facetValue: DateFacetValue) {
+    try {
+      const startDate = deserializeRelativeDate(facetValue.start);
+      const relativeDate =
+        startDate.period === 'past'
+          ? startDate
+          : deserializeRelativeDate(facetValue.end);
+
+      return this.bindings.i18n.t(
+        `${relativeDate.period}-${relativeDate.unit}`,
+        {
+          count: relativeDate.amount,
+        }
+      );
+    } catch (error) {
+      return this.bindings.i18n.t('to', {
+        start: parseDate(facetValue.start).format('YYYY-MM-DD'),
+        end: parseDate(facetValue.end).format('YYYY-MM-DD'),
+      });
+    }
+  }
+  private renderValues() {
+    return this.renderValuesContainer(
+      this.valuesToRender.map((value) => this.renderValue(value))
+    );
+  }
+
+  private renderValue(facetValue: DateFacetValue) {
+    const displayValue = this.formatFacetValue(facetValue);
+    const isSelected = facetValue.state === 'selected';
+    const isExcluded = facetValue.state === 'excluded';
+    return (
+      <FacetValueLink
+        displayValue={displayValue}
+        isSelected={isSelected}
+        numberOfResults={facetValue.numberOfResults}
+        i18n={this.bindings.i18n}
+        onClick={() => this.facet.toggleSingleSelect(facetValue)}
+      >
+        <FacetValueLabelHighlight
+          displayValue={displayValue}
+          isSelected={isSelected}
+          isExcluded={isExcluded}
+        ></FacetValueLabelHighlight>
+      </FacetValueLink>
+    );
+  }
+
+  private renderValuesContainer(children: VNode[]) {
+    return (
+      <FacetValuesGroup i18n={this.bindings.i18n} label={this.displayName}>
+        <ul class="mt-3" part="values">
+          {children}
+        </ul>
+      </FacetValuesGroup>
+    );
+  }
+
+  private renderHeader() {
+    return (
+      <FacetHeader
+        i18n={this.bindings.i18n}
+        label={this.displayName}
+        onClearFilters={() => {
+          this.focusTarget.focusAfterSearch();
+          if (this.hasInputRange) {
+            this.resetRange();
+            return;
+          }
+          this.facet.deselectAll();
+        }}
+        numberOfActiveValues={this.numberOfSelectedValues}
+        isCollapsed={this.isCollapsed}
+        headingLevel={0}
```

Comment:
Mmm interesting, should this maybe be a prop on the atomic-commerce-facets component that would be passed down to the generated facet components?

I presume implementers will typically want all facet headers to have the same heading level, so having a single prop to control them all would probably work fine.